lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -685,37 +685,46 @@ class DecomposeAtenTriuOp : public OpRewritePattern<AtenTriuOp> {
       return rewriter.notifyMatchFailure(op, "the rank of tensor should >= 2");
     }
 
-    auto baseType = ValueTensorType::getWithLeastStaticInformation(context);
     Value cstZero =
         rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(0));
     Value cstOne =
         rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
     Value none = rewriter.create<ConstantNoneOp>(loc);
 
-    Value rowDim = rewriter.create<Torch::ConstantIntOp>(
-        loc, rewriter.getI64IntegerAttr(-2));
-    Value colDim = rewriter.create<Torch::ConstantIntOp>(
-        loc, rewriter.getI64IntegerAttr(-1));
-    Value rowSize = rewriter.create<AtenSizeIntOp>(loc, input, rowDim);
-    Value colSize = rewriter.create<AtenSizeIntOp>(loc, input, colDim);
+    Value rowSize = getTensorDimSize(rewriter, input, -2);
+    Value colSize = getTensorDimSize(rewriter, input, -1);
 
+    auto si64Type = IntegerType::get(context, 64, IntegerType::Signed);
     Value rowArange = rewriter.create<AtenArangeOp>(
-        loc, baseType, rowSize, /*dtype=*/none, /*layout=*/none,
+        loc, getTensorTypeFromValueVector({rowSize}, si64Type), rowSize,
+        /*dtype=*/none, /*layout=*/none,
```

Comment:
If you want, you could change the IntegerType::get to a rewriter.getIntegerType like:
```cpp
    auto si64Type = rewriter.getIntegerType(/*width=*/64, /*isSigned*/=true);
```
Also, it would be good to add the dtype specification to the AtenArangeOp:
```cpp
    auto int64DtypeInt = getDtypeIntValueForType(
        rewriter, loc, si64Type);
```
and specify ```/*dtype=*/int64DtypeInt```.

Reply:
I have made some modifications to the code, please help me review it again.